src/components/Courses/Courses.jsx

```diff
@@ -20,45 +24,73 @@ import {
 	CoursesList,
 	CoursesMessage,
 	CoursesStyled,
+	LoaderStyled,
 } from './Courses.styled';
 
 const Courses = () => {
+	const dispatch = useDispatch();
 	const navigate = useNavigate();
+
+	const initDataFetched = useRef(false);
 	const [searchQuery, setSearchQuery] = useState('');
+	const [isLoading, setIsLoading] = useState(false);
+
+	const { role } = useSelector(selectUser);
+	const isAdmin = role === ROLES.ADMIN;
 
 	const courses = useSelector((state) =>
 		selectCoursesBySearchQuery(state, searchQuery)
 	);
 
+	useEffect(() => {
+		if (initDataFetched.current) return;
+		initDataFetched.current = true;
+
+		(async () => {
+			setIsLoading(true);
+
+			await dispatch(fetchAuthors());
+			await dispatch(fetchCourses());
+
+			setIsLoading(false);
+		})();
+	}, [dispatch]);
+
 	return (
 		<CoursesStyled>
 			<Container>
 				<CoursesHeader>
 					<SearchBar onSubmit={setSearchQuery} />
-					<Button
-						type={ADD_NEW_COURSE_BTN.type}
-						text={ADD_NEW_COURSE_BTN.text}
-						onClick={() => navigate(ROUTES.CREATE_COURSE)}
-					/>
-				</CoursesHeader>
-				{courses.length > 0 ? (
-					<CoursesList>
-						{courses.map((item) => (
-							<li key={item.id}>
-								<CourseCard {...item} />
-							</li>
-						))}
-					</CoursesList>
-				) : (
-					<CoursesMessage>
-						<p>{COURSES_NO_RESULTS_TEXT}</p>
-						<img
-							src={noResults}
-							alt={NO_RESULTS_ALTERNATIVE_TEXT}
-							width={300}
+					{isAdmin && (
+						<Button
+							type={ADD_NEW_COURSE_BTN.type}
+							text={ADD_NEW_COURSE_BTN.text}
+							onClick={() => {
+								navigate(ROUTES.CREATE_COURSE);
+							}}
 						/>
-					</CoursesMessage>
-				)}
+					)}
+				</CoursesHeader>
+				{isLoading && <LoaderStyled />}
+				{!isLoading &&
+					(courses.length > 0 ? (
```

Comment:
For better readability create a new constant.
E.g canRenderCourseList.
I don't recommend use in JSX 2-leveled conditions.

Reply:
Edits to rendering conditions have been added